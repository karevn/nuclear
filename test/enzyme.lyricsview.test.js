import React from 'react';
import { shallow } from 'enzyme';
import { expect } from 'chai';
import { describe, it } from 'mocha';
import sinon from 'sinon';

import { LyricsView } from '../app/components/LyricsView';

describe('<LyricsView />', () => {
  const t = () => '';

  it('Test if there is no track selected', () => {
    const spy = sinon.spy(LyricsView.prototype, 'renderNoSelectedTrack');
    const wrapper = shallow(<LyricsView track={null} t={t} />);
    const instance = wrapper.instance();

    expect(spy.calledOnce).to.be.true;
    expect(wrapper.contains(instance.renderNoSelectedTrack())).to.be.true;
  });
});
