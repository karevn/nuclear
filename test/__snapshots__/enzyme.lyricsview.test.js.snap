// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LyricsView /> Test if there is no track selected 1`] = `
<div
  className="lyrics_view"
>
  <div
    className="empty_state"
  >
    <FontAwesome
      name="music"
    />
    <h2 />
    <div />
  </div>
</div>
`;

exports[`<LyricsView /> Test if there is selected track 1`] = `
<div
  className="lyrics_view"
>
  <LyricsHeader
    artist="ARTIST"
    name="NAME"
  />
  <div
    className="lyrics_text"
  />
</div>
`;
